src/components/pwaBadge.tsx

```diff
@@ -0,0 +1,104 @@
+import { AlertDialog, AlertDialogBody, AlertDialogCloseButton, AlertDialogContent, AlertDialogFooter, AlertDialogHeader, AlertDialogOverlay, Button } from '@chakra-ui/react';
+import { useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+import { If, Then } from 'react-if';
+// eslint-disable-next-line import/no-unresolved
+import { useRegisterSW } from 'virtual:pwa-register/react'
+
+
+export const PWABadge = () => {
+    const period = 1000 * 60 * 2; //ms
+
+    const {
+        offlineReady: [offlineReady, setOfflineReady],
+        needRefresh: [needRefresh, setNeedRefresh],
+        updateServiceWorker,
+    } = useRegisterSW({
+        onRegisteredSW(swUrl, r) {
+            if (period <= 0) return
+            if (r?.active?.state === 'activated') {
+                registerPeriodicSync(period, swUrl, r)
+            }
+            else if (r?.installing) {
+                r.installing.addEventListener('statechange', (e) => {
+                    const sw = e.target as ServiceWorker
+                    if (sw.state === 'activated')
+                        registerPeriodicSync(period, swUrl, r)
+                })
+            }
+        },
+    })
+
+    function onClose() {
+        setOfflineReady(false)
+        setNeedRefresh(false)
+    }
+
+    const cancelRef = useRef<HTMLButtonElement | null>(null);
+    const { t } = useTranslation();
+
+    return (<AlertDialog
```

Comment:
You can use our chackraConfirmation component.

Reply:
done